src/Mod/Path/PathScripts/PathArray.py

```diff
@@ -52,6 +54,12 @@ def __init__(self, obj):
                         "Path", "The number of copies in Linear 1D and Polar pattern")
         obj.addProperty("App::PropertyVector", "Centre",
                         "Path", "The centre of rotation in Polar pattern")
+        obj.addProperty("App::PropertyBool", "SwapDirection",
+                        "Path", "Make copies in X direction before Y in Linear 2D pattern")
+        obj.addProperty("App::PropertyInteger", "JitterPercent",
+                        "Path", "Percent of copies to randomly offset")
+        obj.addProperty("App::PropertyVectorDistance", "JitterMagnitude",
+                        "Path", "Maximum random offset of copies")
         obj.addProperty("App::PropertyLink", "ToolController",
                         "Path", QtCore.QT_TRANSLATE_NOOP("App::Property", "The tool controller that will be used to calculate the path"))
```

Comment:
Does the array really need its own TC reference?  Under what circumstances would the user change the TC in the array and not in the underlying operations?  If the op really needs a local copy but the user shouldn't change it, please hide the property from the user.

Reply:
I think that makes sense to control the Array's TC by the underlying operations.  Will hide.